sdk/core/azure-core/inc/azure/core/http/win_http_transport.hpp

```diff
@@ -134,6 +134,27 @@ namespace Azure { namespace Core { namespace Http {
     bool IgnoreUnknownCertificateAuthority = false;
   };
 
+  namespace _internal {
+    /**
+     * @brief Factory to produce a context with special settings like no client cert context.
+     *
+     * @details Client SDKs can use this class to create an Azure Context to change the
+     * WinHttpTransport default behavior. This is because a Client SDK doesn't know about the
+     * specifics of a transport adapter. So, on Windows, SDK client can set options and only
+     * WinHTTPTransport will check for those options to be in the context.
+     *
+     */
+    class WinHttpTransportContextProvider {
+    public:
+      // Factory with no constructor
+      WinHttpTransportContextProvider() = delete;
+
+      // Creates a set up token to make
+      static Azure::Core::Context GetNoClientCertificateContext(Azure::Core::Context const& parent);
+      static bool HasNoClientCertificateConfiguration(Azure::Core::Context const& context);
+    };
+  } // namespace _internal
+
```

Comment:
Can we just define a magic string here like NoClientCertificate = "blahblah"?

The policy will set a bool value with this string and we check this value in WinHTTP transport layer. It's much simpler

Reply:
Using a magic string creates a hard dependency between two components:  The one setting the option and the one reading it. 
So yes, it would be much simpler, but it costs `tech debt`.

The dependency is broken when we use a factory to hide what's the actual key (the magic string) and how to look for it.

Reply:
How current implementation better than a magic string? A string is just a symbol attestation will reference. A static function name is also a symbol.